### ISSUE TITLE
Enable email user registration

### ISSUE THREAD

**Issue by Callustian** [ARETEedu/moodle-mod_arete#81]
As the instance is at this time it is not possible for a user to register and enter the site. It is necessary to enable email user registration.

**Comment by fwild** [comment]
Hm. Moodle provides registration and email verification already, why write a new one?

**Comment by Callustian** [comment]
I understand this feature is disabled and we want to enable this feature. The task is not about implement email user registration, the task is about persisting the setting. Am I wrong about the requirement? Thanks

**Comment by fwild** [comment]
No, the self-registration feature is activated on all sites, and I do not think it is allowed to persist it from within a single plugin. (Would be rather counter intuitive from a Moodle policy perspective). The request was about checking if there is a possibility to use a web service directly to execute the self registration from within the app via an API -- without having to send the user on to the web to do the registration. This feature is relevant only for sites that have email self registration enabled - not relevant for more closed sites (like a university Moodle typically would be).

**Comment by Callustian** [comment]
From ( https://moodle.org/mod/forum/discuss.php?d=356702 ) This discussion states that you can  
`curl -X POST \
  '[https://my.moodleapp.tld/login/token.php?service=moodle_mobile_app'](https://my.moodleapp.tld/login/token.php?service=moodle_mobile_app%27) \
  -d 'username=mymoodleuser&password=MySecretPassword'`
And do the operations you necessitate. The setting has changed a bit, it is in server. I have tried this and seems to generate the token.

**Comment by fwild** [comment]
Bit confused - this is implemented in the MirageXR API - it generates tokens, which then are used to auth the future list/search API calls. My question was about something else: if you do not have an account, you need to self-register one - and for that, currently, we use the Moodle self-registration pages via the web, e.g. https://arete.ucd.ie/login/signup.php

Instead of calling it via html pages, it would be neater, if we had an API call for that instead. But never mind - as said, likely someone built that already and this is more a configuration job (and it can always be hidden from the user by POST from within the app and parsing the resulting html).

**Comment by Callustian** [comment]
I misunderstood the requirements, I apologise.
Something I read people are attempting, is making a new user using ( https://docs.moodle.org/dev/Web_service_API_functions ). This requires the system is already logged in using a privileged account.
I am unsure on how to do the self registration without the use of some client library that does the post (and there is the captcha)

**Comment by fwild** [comment]
ok, closing this ticket accordingly.
